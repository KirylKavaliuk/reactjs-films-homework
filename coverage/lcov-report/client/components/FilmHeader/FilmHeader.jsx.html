<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components/FilmHeader/FilmHeader.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">client/components/FilmHeader</a> FilmHeader.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import withConditionalRendering from 'utils/rendering';
&nbsp;
import styles from './FilmHeader.scss';
&nbsp;
const FilmHeader = ({ name, genres, duration }) =&gt; {
  const absoluteDuration = Math.abs(duration);
  const hours = Math.floor(absoluteDuration / 60);
  const minutes = (absoluteDuration % 60).toFixed();
&nbsp;
  const renderGenres = () =&gt; (
    genres.map((genre, index) =&gt; (
      &lt;span
        key={ index }
        className={ styles.genre }
      &gt;{ genre }&lt;/span&gt;))
  );
&nbsp;
  const setDuration = () =&gt; (
    duration &amp;&amp; &lt;span className={ styles.duration }&gt;
      { +hours === 0 ? null : `${hours} h`} { +minutes === 0 ? null : `${minutes} m` }
    &lt;/span&gt;
  );
&nbsp;
  return (
    &lt;div className={ styles.filmHeader }&gt;
      &lt;h2 className={ styles.name }&gt;{ name }&lt;/h2&gt;
      &lt;div className={ styles.info }&gt;
        { renderGenres() }
        { setDuration() }
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
FilmHeader.propTypes = {
  name: PropTypes.string,
  genres: PropTypes.array,
  duration: PropTypes.number,
};
&nbsp;
FilmHeader.defaultProps = {
  name: '',
  duration: 0,
};
&nbsp;
export default withConditionalRendering(FilmHeader, 'genres');
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 01 2019 11:31:09 GMT+0300 (GMT+03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
