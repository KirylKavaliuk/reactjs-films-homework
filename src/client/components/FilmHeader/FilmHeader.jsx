import React from 'react';

import PropTypes from 'prop-types';

import styles from './FilmHeader.scss';

const FilmHeader = ({ name, genres, duration }) => {
  const absoluteDuration = Math.abs(duration);
  const hours = Math.floor(absoluteDuration / 60);
  const minutes = (absoluteDuration % 60).toFixed();

  return (
    <div className={ styles.filmHeader }>
      <h2 className={ styles.name }>{ name }</h2>
      <p className={ styles.genres }>
        { genres.map((genre, index) => (
          <span
            key={ index }
            className={ styles.genre }
          >{ genre }</span>
        )) }
        { !!duration && <span className={ styles.duration }>
            { +hours === 0 ? null : `${hours} h`} { +minutes === 0 ? null : `${minutes} m` }
          </span>
        }
      </p>
    </div>
  );
};

FilmHeader.propTypes = {
  name: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  duration: PropTypes.number.isRequired,
};

export default FilmHeader;
