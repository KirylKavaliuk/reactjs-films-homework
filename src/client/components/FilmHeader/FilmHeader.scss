.film-header {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  width: 100%;
  max-width: 500px;
  align-items: flex-start;
  grid-area: film-header;
  align-self: end;

  &:hover {
    cursor: default;
  }
}

.name {
  box-sizing: border-box;
  padding: 10px 0;
  text-transform: uppercase;
  font-size: 50px;
  font-weight: 700;
}

.info {
  display: flex;
  max-width: 450px;
  line-height: 1.4;
  justify-content: flex-start;
  flex-wrap: wrap;
}

.genre {
  box-sizing: border-box;
  padding-right: 25px;
  font-size: 18px;
}

.duration {
  font-size: 18px;

  &::before {
    padding-right: 20px;
    content: '|';
  }
}

@media only screen and (max-width: 850px) {
  .name {
    font-size: 35px;
  }

  .genre {
    font-size: 13px;
  }

  .duration {
    font-size: 13px !important;
  }
}
