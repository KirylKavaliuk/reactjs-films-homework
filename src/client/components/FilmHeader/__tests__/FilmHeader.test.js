import React from 'react';
import renderer from 'react-test-renderer';
import FilmHeader from '../FilmHeader';

it('<FilmHeader/>', () => {
  const tree = renderer.create(
    <FilmHeader genres={ ['genre1', 'genre2', 'genre3'] }/>,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<FilmHeader/>', () => {
  const tree = renderer.create(
    <FilmHeader
      genres={ ['genre1', 'genre2', 'genre3'] }
      duration={ 250 }
    />,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<FilmHeader/>', () => {
  const tree = renderer.create(
    <FilmHeader
      genres={['genre1', 'genre2', 'genre3']}
      duration={ 60 }
    />,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<FilmHeader/>', () => {
  const tree = renderer.create(
    <FilmHeader
      genres={['genre1', 'genre2', 'genre3']}
      duration={ 50 }
    />,
  ).toJSON();

  expect(tree).not.toBeNull();
  expect(tree).toMatchSnapshot();
});

it('<FilmHeader/>', () => {
  const tree = renderer.create(<FilmHeader/>).toJSON();

  expect(tree).toBeNull();
  expect(tree).toMatchSnapshot();
});
