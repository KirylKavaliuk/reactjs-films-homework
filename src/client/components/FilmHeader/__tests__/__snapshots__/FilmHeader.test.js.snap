// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders film header correctly 1`] = `
<div
  className="FilmHeader__film-header___1bjD8"
>
  <h2
    className="FilmHeader__name___12D1E"
  >
    
  </h2>
  <p
    className="FilmHeader__genres___2kuJ_"
  >
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre1
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre2
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre3
    </span>
    0
  </p>
</div>
`;

exports[`renders film header correctly 2`] = `
<div
  className="FilmHeader__film-header___1bjD8"
>
  <h2
    className="FilmHeader__name___12D1E"
  >
    
  </h2>
  <p
    className="FilmHeader__genres___2kuJ_"
  >
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre1
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre2
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre3
    </span>
    <span
      className="FilmHeader__duration___1HHqc"
    >
      4 h
       
      10 m
    </span>
  </p>
</div>
`;

exports[`renders film header correctly 3`] = `
<div
  className="FilmHeader__film-header___1bjD8"
>
  <h2
    className="FilmHeader__name___12D1E"
  >
    
  </h2>
  <p
    className="FilmHeader__genres___2kuJ_"
  >
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre1
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre2
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre3
    </span>
    <span
      className="FilmHeader__duration___1HHqc"
    >
      1 h
       
    </span>
  </p>
</div>
`;

exports[`renders film header correctly 4`] = `
<div
  className="FilmHeader__film-header___1bjD8"
>
  <h2
    className="FilmHeader__name___12D1E"
  >
    
  </h2>
  <p
    className="FilmHeader__genres___2kuJ_"
  >
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre1
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre2
    </span>
    <span
      className="FilmHeader__genre___2QZId"
    >
      genre3
    </span>
    <span
      className="FilmHeader__duration___1HHqc"
    >
       
      50 m
    </span>
  </p>
</div>
`;

exports[`renders film header uncorrectly 1`] = `null`;
